packages/edit-post/src/components/sidebar/settings-header/index.js

```diff
@@ -3,54 +3,84 @@
-	const blockLabel = __( 'Block' );
-	const [ documentAriaLabel, documentActiveClass ] = sidebarName === 'edit-post/document' ?
+	const [ documentAriaLabel, documentSettingsActive ] = sidebarName === 'edit-post/document' ?
 		// translators: ARIA label for the Document sidebar tab, selected.
-		[ __( 'Document (selected)' ), 'is-active' ] :
+		[ __( 'Document (selected)' ), true ] :
 		// translators: ARIA label for the Document sidebar tab, not selected.
-		[ __( 'Document' ), '' ];
+		[ __( 'Document' ), false ];
 
-	const [ blockAriaLabel, blockActiveClass ] = sidebarName === 'edit-post/block' ?
+	const [ blockAriaLabel, blockSettingsActive ] = sidebarName === 'edit-post/block' ?
 		// translators: ARIA label for the Settings Sidebar tab, selected.
-		[ __( 'Block (selected)' ), 'is-active' ] :
+		[ __( 'Block (selected)' ), true ] :
 		// translators: ARIA label for the Settings Sidebar tab, not selected.
-		[ __( 'Block' ), '' ];
+		[ __( 'Block' ), false ];
+
+	const documentSettingsName = 'edit-post-settings__panel-tab-document';
+	const blockSettingsName = 'edit-post-settings__panel-tab-block';
+
+	const [ selectedTabName, setSelectedTabName ] = useState( documentSettingsName );
+
+	const tabs = [
+		{
+			name: documentSettingsName,
+			className: 'edit-post-sidebar__panel-tab',
+			title: __( 'Document' ),
+			ariaLabel: `${ documentAriaLabel }`,
+		},
+		{
+			name: blockSettingsName,
+			className: 'edit-post-sidebar__panel-tab',
+			title: __( 'Block' ),
+			ariaLabel: `${ blockAriaLabel }`,
+		},
+	];
+
+	const onSelect = ( tabName ) => {
+		setSelectedTabName( tabName );
+		if ( tabName === documentSettingsName ) {
+			openDocumentSettings();
+		} else if ( tabName === blockSettingsName ) {
+			openBlockSettings();
+		}
+	};
+
+	useEffect(
+		() => {
+			if ( documentSettingsActive ) {
+				setSelectedTabName( documentSettingsName );
+			}
+		}, [ documentSettingsActive ]
+	);
+
+	useEffect(
+		() => {
+			if ( blockSettingsActive ) {
+				setSelectedTabName( blockSettingsName );
+			}
+		}, [ blockSettingsActive ]
+	);
 
 	return (
 		<SidebarHeader
 			className="edit-post-sidebar__panel-tabs"
 			closeLabel={ __( 'Close settings' ) }
 		>
-			{ /* Use a list so screen readers will announce how many tabs there are. */ }
-			<ul>
-				<li>
-					<button
-						onClick={ openDocumentSettings }
-						className={ `edit-post-sidebar__panel-tab ${ documentActiveClass }` }
-						aria-label={ documentAriaLabel }
-						data-label={ __( 'Document' ) }
-					>
-						{ __( 'Document' ) }
-					</button>
-				</li>
-				<li>
-					<button
-						onClick={ openBlockSettings }
-						className={ `edit-post-sidebar__panel-tab ${ blockActiveClass }` }
-						aria-label={ blockAriaLabel }
-						data-label={ blockLabel }
-					>
-						{ blockLabel }
-					</button>
-				</li>
-			</ul>
+			<TabPanel
+				tabs={ tabs }
+				onSelect={ onSelect }
+				controlledTabName={ selectedTabName }
+			>
+				{ () => { } }
```

Comment:
@jffng Hallooo! This PR looks great! I was a little confused at this one (at first glance).

What do you think about adjusted the original `TabPanel` component's `children()` render to do something like...

`children && children(selectedTab)`

That way, you won't have to pass it in a children render function that returns null/empty.

P.S. I'm not sure what the current conventions are for handling this flow 😊. The flow I suggested feels little cleaner + less ambiguous

Reply:
Thanks @ItsJonQ! I agree. We could also set `children`'s default value to `noop`.

Reply:
Aww yiss! `noop` it up! 🎉